### ISSUE TITLE
[New rule] Avoid return FutureOr

### ISSUE THREAD

**Issue by incendial** [dart-code-checker/dart-code-metrics#114]
**Please describe what the rule should do:**
Warns when return type is `FutureOr<T>`, because end users need to check whether get back a T or a Future<T> before they can do anything useful.

**What category of rule is this? (place an "X" next to just one item)**

[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Bad:
```dart
FutureOr<int> triple(FutureOr<int> value) {
  if (value is int) return value * 3;
  return (value as Future<int>).then((v) => v * 3);
}
```

Good:
```dart
Future<int> triple(FutureOr<int> value) async => (await value) * 3;
```

**Are you willing to submit a pull request to implement this rule?**
Yes
